modules/images/dominant-color/load.php

```diff
@@ -126,8 +126,12 @@ function dominant_color_img_tag_add_dominant_color( $filtered_image, $context, $
 		$extra_class  = $image_meta['has_transparency'] ? 'has-transparency' : 'not-transparent';
 	}
 
-	if ( ! empty( $data ) || ! empty( $style ) ) {
-		$filtered_image = str_replace( '<img ', '<img ' . $data . $style, $filtered_image );
+	if ( ! empty( $data ) || ! empty( $style ) || ! str_contains( $data, $filtered_image ) ) {
+		if ( str_contains( $filtered_image, 'style=' ) ) {
+			$filtered_image = str_replace( 'style="', $data . 'style="' . $style, $filtered_image );
+		} else {
+			$filtered_image = str_replace( '<img ', '<img ' . $data . ' style="' . $style . '" ', $filtered_image );
```

Comment:
See my overarching PR comment, we should avoid adding an empty style attribute.

Also, if `$data` is not empty, it always has a space in the end already, so let's avoid double spaces here.
```suggestion
		} elseif ( ! empty( $style ) ) {
			$filtered_image = str_replace( '<img ', '<img ' . $data . 'style="' . $style . '" ', $filtered_image );
		} else {
			$filtered_image = str_replace( '<img ', '<img ' . $data, $filtered_image );
```